rust/private/rustdoc.bzl

```diff
@@ -119,15 +119,16 @@ def _zip_action(ctx, input_dir, output_zip):
         output_zip (File): The location of the output archive containing generated documentation
     """
     args = ctx.actions.args()
-
-    # Create but not compress.
-    args.add("c", output_zip)
+    args.add("--zipper", ctx.executable._zipper)
+    args.add("--output", output_zip)
+    args.add("--root-dir", input_dir.path)
```

Comment:
Should this be the input_dir.path, rather than the workspace root?  It seems like you should be able to unzip all of your docs files into a single folder and have them laid out appropriately.

I'm not super familiar with the structure of rust_docs pages, so feel free to correct me if I'm wrong on that.

Reply:
Hmm; I’m not sure that I understand the question?

The directory layout at this stage of the Bazel action is like:

```
bazel-bin/k8-fastbuild/path/to/target/LICENSE.txt
bazel-bin/k8-fastbuild/path/to/target/favicon.ico
bazel-bin/k8-fastbuild/path/to/target/crate_name/index.html
bazel-bin/k8-fastbuild/path/to/target/crate_name/struct.Foo.html
bazel-bin/k8-fastbuild/path/to/target/src/mylib/lib.rs.html
```

and we want to create a zip archive that has files

```
LICENSE.txt
favicon.ico
crate_name/index.html
crate_name/struct.Foo.html
src/mylib/lib.rs.html
```

so that we can point a web server inside this zip file to serve docs.
The value of `input_dir` is `bazel-bin/k8-fastbuild/path/to/target`, so
we do so by stripping off `input_dir.path`.

Is the comment that we instead want to preserve one level of directory
in the output zip, like this?

```
target/LICENSE.txt
target/favicon.ico
target/crate_name/index.html
target/crate_name/struct.Foo.html
target/src/mylib/lib.rs.html
```

If so, this is a reasonable suggestion, and I can make that change if
you like.

Reply:
Thanks for the clear example.  I was attempting to suggest going from:
```
bazel-bin/k8-fastbuild/path/to/target/LICENSE.txt
bazel-bin/k8-fastbuild/path/to/target/favicon.ico
bazel-bin/k8-fastbuild/path/to/target/crate_name/index.html
bazel-bin/k8-fastbuild/path/to/target/crate_name/struct.Foo.html
bazel-bin/k8-fastbuild/path/to/target/src/mylib/lib.rs.html
```

to
```
path/to/target/LICENSE.txt
path/to/target/favicon.ico
path/to/target/crate_name/index.html
path/to/target/crate_name/struct.Foo.html
path/to/target/src/mylib/lib.rs.html
```

which, in my head, would preserve the aggregation of all rustdoc zip files within any given bazel project into a single directory.  Does that make more sense?

Reply:
Got it. I think I can work with that. Will address (not done yet).

Reply:
Done.